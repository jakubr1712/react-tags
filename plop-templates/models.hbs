export interface I{{properCase name}}ComponentProps {}