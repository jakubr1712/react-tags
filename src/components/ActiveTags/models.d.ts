export interface IActiveTagsProps {
  tags: CommonTypes.ITag[];
  removeTag: (tagId: number) => void;
}