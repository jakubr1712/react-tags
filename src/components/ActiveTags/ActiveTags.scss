.active-tags {
  display: flex;
  flex-wrap: wrap;
  gap: 8px;
  margin-top: 16px;
}
