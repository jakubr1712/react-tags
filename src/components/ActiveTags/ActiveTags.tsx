import React from "react";
import "./ActiveTags.scss";
import Tag from "../Tag/Tag";
import { IActiveTagsProps } from "./models.d";

const ActiveTags: React.FC<IActiveTagsProps> = ({ tags, removeTag }) => {
  const tagsAll = tags
    .filter((tag) => tag.isActive)
    .map((tag) => (
      <Tag
        key={tag.id}
        id={tag.id}
        name={tag.name}
        onRemove={() => removeTag(tag.id)}
      />
    ));

  return <div className="active-tags">{tagsAll}</div>;
};

export default ActiveTags;
