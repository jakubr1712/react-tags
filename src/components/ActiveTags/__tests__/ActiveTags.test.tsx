import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import ActiveTags from "../ActiveTags";
import { IActiveTagsProps } from "../models.d";

describe("ActiveTags component", () => {
  const mockRemoveTag = jest.fn();

  const defaultProps: IActiveTagsProps = {
    tags: [
      { id: 1, name: "Tag1", count: 10, isActive: true },
      { id: 2, name: "Tag2", count: 20, isActive: true },
    ],
    removeTag: mockRemoveTag,
  };

  beforeEach(() => {
    mockRemoveTag.mockClear();
  });

  it("renders properly and matches snapshot", () => {
    const { container } = render(<ActiveTags {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders all active tags", () => {
    render(<ActiveTags {...defaultProps} />);
    expect(screen.getByText("Tag1")).toBeInTheDocument();
    expect(screen.getByText("Tag2")).toBeInTheDocument();
  });

  it("calls removeTag with correct id when remove button is clicked", () => {
    render(<ActiveTags {...defaultProps} />);
    fireEvent.click(screen.getAllByText("×")[0]);
    expect(mockRemoveTag).toHaveBeenCalledWith(1);

    fireEvent.click(screen.getAllByText("×")[1]);
    expect(mockRemoveTag).toHaveBeenCalledWith(2);
  });
});
