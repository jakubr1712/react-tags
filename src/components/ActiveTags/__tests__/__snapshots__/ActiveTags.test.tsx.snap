// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveTags component renders properly and matches snapshot 1`] = `
<div>
  <div
    class="active-tags"
  >
    <div
      class="tag"
    >
      Tag1
      <button
        class="tag__remove-button"
      >
        ×
      </button>
    </div>
    <div
      class="tag"
    >
      Tag2
      <button
        class="tag__remove-button"
      >
        ×
      </button>
    </div>
  </div>
</div>
`;
